conductr_cli/bundle_core_info.py

```diff
@@ -0,0 +1,49 @@
+from conductr_cli.conduct_info_inspect import has_bundle_id, has_bundle_name
+
+
+class BundleCoreInfo(object):
+    def __init__(self, bundle_id, bundle_name, bundle_digest, configuration_digest):
+        self.bundle_id = bundle_id
+        self.bundle_name = bundle_name
+        self.bundle_digest = bundle_digest
+        self.configuration_digest = configuration_digest
+        self.bundle_name_with_digest = str.format(
+            "{0}-{1}", bundle_name, bundle_digest)
+
+    @classmethod
+    def from_bundles(cls, bundles_json):
+        return list(map(
+            lambda b: cls(bundle_id=b["bundleId"],
+                          bundle_digest=b["bundleDigest"],
+                          configuration_digest=b.get("configurationDigest", ""),
+                          bundle_name=b["attributes"]["bundleName"]),
+            bundles_json))
+
+    @classmethod
+    def filter_by_bundle_id(cls, bundle_core_info, bundle_id):
+        for bundle_info in bundle_core_info:
+            if has_bundle_id(bundle_info.bundle_id, bundle_id) or has_bundle_name(bundle_info.bundle_name, bundle_id):
+                return bundle_info
+        return None
+
+    @staticmethod
+    def diff(this, that):
+        return set(this) - set(that)
+
+    def __str__(self) -> str:
+        return str.format("{0} - {1}", self.bundle_id, self.bundle_name)
+
+    def __repr__(self):
+        return "Item(%s)" % self.bundle_id
+
+    def __eq__(self, other):
+        if isinstance(other, BundleCoreInfo):
+            return self.bundle_id == other.bundle_id
```

Comment:
Is there a reason why we compare only the bundle id?

The reason why I'm asking is because if we were to compare all the fields, we can use equality (and hence `assertEqual`) in our tests.

```
    def __eq__(self, other):
        return self.__dict__ == other.__dict__ if isinstance(other, self.__class__) else False
```

This is used in various places in our code: https://github.com/typesafehub/conductr-cli/blob/master/conductr_cli/license_validation.py#L31